addons/account/models/account_analytic_line.py

```diff
@@ -80,50 +83,78 @@ def action_view_vendor_bill(self):
         }
         return result
 
-class AccountAnalyticTag(models.Model):
-    _inherit = 'account.analytic.tag'
-
-    @api.constrains('company_id')
-    def _check_company_consistency(self):
-        analytic_tags = self.filtered('company_id')
-
-        if not analytic_tags:
-            return
 
-        self.flush_recordset(['company_id'])
-        self.env['account.move.line'].flush_model(['company_id'])
-        self._cr.execute('''
-            SELECT line.id
-            FROM account_analytic_tag_account_move_line_rel tag_rel
-            JOIN account_analytic_tag tag ON tag.id = tag_rel.account_analytic_tag_id
-            JOIN account_move_line line ON line.id = tag_rel.account_move_line_id
-            WHERE tag_rel.account_analytic_tag_id IN %s
-            AND line.company_id != tag.company_id
-        ''', [tuple(analytic_tags.ids)])
+class AccountAnalyticPlan(models.Model):
+    _inherit = 'account.analytic.plan'
 
-        if self._cr.fetchone():
-            raise UserError(_("You can't set a different company on your analytic tags since there are some journal items linked to it."))
+    def _get_applicability(self, **kwargs):
+        # OVERRIDE analytic
+        self.ensure_one()
+        if 'applicability' in kwargs:
+            return kwargs['applicability']
+        domain = kwargs.get('domain', None)
+        product = self.env['product.product'].browse(kwargs.get('product', None))
+        account = self.env['account.account'].browse(kwargs.get('account', None))
+        if not domain and not product and not account:
+            return self.default_applicability
+        best_applicability = {'applicability': None, 'score': 0}
+        for rule in self.applicability_ids:
+            rule_score = 0
+            if domain and rule.domain == domain:
+                rule_score += 1
+            if domain and rule.domain != domain:
+                continue
+            if rule.account_prefix and account and account.code.startswith(rule.account_prefix):
+                rule_score += 1
+            elif rule.account_prefix and account and not account.code.startswith(rule.account_prefix):
+                continue
+            if rule.product_categ_id and product and rule.product_categ_id == product.categ_id:
+                rule_score += 1
+            elif rule.product_categ_id and product and rule.product_categ_id != product.categ_id:
+                continue
```

Comment:
So there is a priority of weight in the different fields?

Reply:
No, there shouldn't be. As soon as a value doesn't correspond, we should not care about this applicability rule.